compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1465,8 +1465,8 @@ class Typer extends Namer
       case _ =>
         if tree.isInline then checkInInlineContext("inline match", tree.srcPos)
         val sel1 = typedExpr(tree.selector)
-        val selType = fullyDefinedType(sel1.tpe, "pattern selector", tree.span).widen
-
+        val rawSelectorTpe = fullyDefinedType(sel1.tpe, "pattern selector", tree.span)
+        val selType = if (tree.isInline) rawSelectorTpe.widenTermRefExpr else rawSelectorTpe.widen
```

Comment:
I am just guessing here, no idea if it's the right solution. widening seems a bit of a magic to me and not sure why we need to do it here and in what exact cases we need to do that :/